### PR TITLE
[41] [Integrate] As a user , I can reset my password

### PR DESCRIPTION
**Pull request by manh-t**
https://github.com/manh-t/flutter-survey/issues/41

## What happened 👀

When call API POST forgot password successfully, a notification snackbar show up.

## Proof Of Work 📹

https://user-images.githubusercontent.com/60863885/126433994-14d41261-c919-4529-9f5e-b82e4c9bca50.mp4

### CODE REVIEW

**Review by markgravity — APPROVED** [review]
LGTM 🤖

**Comment by sleepylee** [comment]
please update the POW 🐰

**Comment by manh-t** [comment]
@sleepylee on it 🏃

**Comment by manh-t** [comment]
@sleepylee Updated PoW 🙇

File: lib/pages/forgotpassword/forgot_password_model.dart [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  void submit() {
+    state = ForgotPasswordState.loading();
+    _resetPasswordUseCase.call(emailController.text).then((result) {
+      if (result is Success) {
+        emailController.clear();
+        state = ForgotPasswordState.success();
+      } else {
+        state = ForgotPasswordState.error(NetworkExceptions.getErrorMessage(
+            (result as Failed).exception.networkExceptions));
```
Comment (sleepylee) [inline lib/pages/forgotpassword/forgot_password_model.dart:23]: how do you think about this way of writing:

```
if (result is Success) {
   // do xyz
}

if (result is Failed) {
  // then you don't need to cast (result as Failed)
}

```

**Review by sleepylee — APPROVED** [review]

**Comment by manh-t** [comment]
@sleepylee thank for the suggestion, updated 🙆‍♂️

**Pull request merged**